environments/ruby/server.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+require 'rack'
+
+CODEPATH = '/userfunc/user'
+
+module Fission
+  class Request < Rack::Request
+    def headers
+      Hash[
+        *env.select { |k,v| k.start_with?('HTTP_') }
+             .map { |k,v| [k.sub(/\AHTTP_/, '').split('_').map(&:capitalize).join('-'), v] }
+             .sort
+             .flatten
+      ]
+    end
+  end
+
+  class Context
+    attr_reader :env
+
+    def initialize(env)
+      @env = env
+    end
+
+    def request
+      @request ||= Request.new(env)
+    end
+  end
+
+  module Specializer
+    def self.call(env)
+      load CODEPATH
+      Rack::Response.new([], 201).finish
+    rescue
```

Comment:
Can you log code loading errors to stdout?  (Otherwise things will just fail mysteriously if there's an error in the user's code.)